perception/sub8_perception/nodes/test_path_marker.py

```diff
@@ -0,0 +1,122 @@
+        middle = pts[0]+(pts[1]-pts[0])/2
+        theta = np.arctan( (float(pts[1][1])-pts[0][1]) / (pts[1][0] - pts[0][0]) )
+        res.pose.x = middle[0]
+        res.pose.y = middle[1]
+        res.pose.theta = theta
+        return res
+
+    def _test_bag(self, filename, duration, pts):
+        rospy.sleep(rospy.Duration(0.5))
+        correct = self.get_ideal_response(pts)
+        print "Correct pose: {}".format(np.degrees(correct.pose.theta))
+        bag = rosbag.Bag(filename)
+        first = True
+        first_time = None
+        previous_time = None
+        i = 0
+        for topic, msg, t in bag.read_messages():
+            if first:
+                first_time = t
+                previous_time = t
+                first = False
+            else:
+                rospy.sleep(t-previous_time) # Slow down to watch happen
+                previous_time = t
+                if t - first_time > duration:
+                    did = True
+                    break
+            if topic == self.img_topic:
+                self.img_pub.publish(msg)
+            if topic == self.info_topic:
+                self.cam_info_pub.publish(msg)
+        res = self.service(VisionRequest2DRequest())
+        # 10 pixel error accepted
+        res_xy = np.array([res.pose.x, res.pose.y])
+        correct_xy = np.array([correct.pose.x, correct.pose.y])
+        err = np.linalg.norm(res_xy-correct_xy)
+        msg="Marker pose (x,y) too much error Res={} Correct={} Error={}".format(res_xy, correct_xy, err)
+        self.assertLess(err, 
+                        10.0,
+                        msg=msg)
+        #5 degrees error accepted
+        theta_err = abs(np.arctan2(np.sin(res.pose.theta-correct.pose.theta), np.cos(res.pose.theta-correct.pose.theta)))
+        msg = "Marker pose angle (theta) too much error Res={} Correct={} Error={}".format(np.degrees(res.pose.theta),
+                                                                                           np.degrees(correct.pose.theta),
+                                                                                           np.degrees(theta_err))
+        self.assertLess(theta_err, 
+                        0.15,
+                        msg=msg)
+
+    def test_transdec(self):
+        print "PARAM", rospy.get_param_names()
+        self._test_bag(rospy.get_param(self.name+'/transdec_path1'),
+                       rospy.Duration(3.10),
+                       [ [43, 443], [216,307] ])
+
+    #Disabled because not long enough
+    #~ def test_transdec2(self):
+        #~ self._test_bag('/home/kallen/bag/path_marker_transdcec/fixed/transdec_path2.bag',
+                       #~ rospy.Duration(1.9),
+                       #~ [ [392, 166], [428,16] ])
+    def test_transdec3(self):
+        self._test_bag(rospy.get_param(self.name+'/transdec_path3'),
+                       rospy.Duration(0.8),
+                       [ [244,9],[321, 127] ])
+    # Disabled because too noisey
```

Comment:
It's so glarry I can barely identify it myself. Hope to get a few more testing bags once new down camera is installed.